File: Intermediate Algorithm Scripting/Python/steamrollArray.py
# Flatten a nested array.
# You must account for varying levels of nesting.

def steamrollArray(arr):
    newArr = []
    for x in arr:
        if type(x) == list:
            newArr.extend(steamrollArray(x))
        else:
            newArr.append(x)
    return newArr

File: Intermediate Algorithm Scripting/Javascript/steamrollArray.js
/*
Flatten a nested array.
You must account for varying levels of nesting.
*/

function steamrollArray(arr) {
  let newArr = [];
  for (let x = 0; x < arr.length; x++) {
    if (Array.isArray(arr[x])) {
      newArr.push(...steamrollArray(arr[x]));
    } else {
      newArr.push(arr[x]);
    }
  }
  return newArr;
}
